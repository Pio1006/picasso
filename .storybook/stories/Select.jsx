import React from 'react'
import { storiesOf } from '@storybook/react'
import { Select } from '../../components'

const OPTIONS = [
  { value: 1, text: 'Item 1' },
  { value: 2, text: 'Item 2' },
  { value: 3, text: 'Item 3' },
  { value: 4, text: 'Item 4' },
  {
    value: 5,
    text: (
      <div>
        Hello <b>world!</b>
      </div>
    )
  }
]

storiesOf('Select', module)
  .add('default', () => <Select label="Select" value={3} options={OPTIONS} />)
  .add('outlined', () => (
    <Select label="Select" value={3} variant="outlined" options={OPTIONS} />
  ))
  .add('outlined open', () => (
    <Select
      label="Select"
      value={3}
      variant="outlined"
      options={OPTIONS}
      open
      onClose={() => {}}
    />
  ))
