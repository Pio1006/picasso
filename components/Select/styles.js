import { PicassoProvider } from '../Picasso'

PicassoProvider.override(({ pallete }) => ({
  MuiSelect: {
    outlined: {
      border: `solid 1px ${pallete.grey[50]}`,
      borderRadius: 0,
      padding: '0.7em 1.5em 0.7em 0.7em',

      '&:focus, &:hover': {
        borderColor: pallete.primary.main,
        backgroundColor: 'transparent'
      }
    },
    selectMenu: {
      minHeight: 'auto',
      lineHeight: '1em',
      fontSize: '16px'
    }
  }
}))

export default {}
