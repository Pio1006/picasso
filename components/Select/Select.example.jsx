import React from 'react'
import { storiesOf } from '@storybook/react'

import Select from './'

const OPTIONS = [
  { value: 1, text: 'Item 1' },
  { value: 2, text: 'Item 2' },
  { value: 3, text: 'Item 3' },
  { value: 4, text: 'Item 4' },
  {
    value: 5,
    text: (
      <div>
        Hello <b>world!</b>
      </div>
    )
  }
]

storiesOf('Select', module)
  .add('default', () => <Select label='Select' options={OPTIONS} value={3} />)
  .add('outlined', () => (
    <Select label='Select' options={OPTIONS} value={3} variant='outlined' />
  ))
  .add('outlined open', () => (
    <Select
      label='Select'
      onClose={() => {}}
      open
      options={OPTIONS}
      value={3}
      variant='outlined'
    />
  ))
