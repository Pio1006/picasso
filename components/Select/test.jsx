import React from 'react'
/* eslint-disable-next-line */
import { render, fireEvent, cleanup } from 'react-testing-library'

import Select from './index'

const renderSelect = (props = {}) => {
  return render(<Select {...props} value='val1' />)
}

const OPTIONS = [
  {
    value: 'val1',
    text: 'text1'
  },
  {
    value: 'val2',
    text: 'text2'
  },
  {
    value: 'val3',
    text: 'text3'
  }
]

afterEach(cleanup)

test('renders native select', () => {
  const { container } = renderSelect({
    native: true,
    options: OPTIONS
  })

  expect(container).toMatchSnapshot()
})

test('renders dropdown select', () => {
  const { container } = renderSelect({
    options: OPTIONS
  })

  expect(container).toMatchSnapshot()
})
