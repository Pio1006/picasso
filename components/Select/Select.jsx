import React from 'react'
import PropTypes from 'prop-types'
import MUISelect from '@material-ui/core/Select'
import { withStyles } from '@material-ui/core/styles'

import OutlinedInput from '../OutlinedInput'
import MenuItem from '../MenuItem'
import styles from './styles'

const renderOptions = props => {
  const { options, native } = props

  if (!options.length) {
    return null
  }

  const OptionComponent = native ? 'option' : MenuItem

  return options.map(({ key, value, text }) => (
    <OptionComponent key={key || value} value={value}>
      {text}
    </OptionComponent>
  ))
}

const Select = props => {
  // eslint-disable-next-line
  const { children, options, variant, classes, inputProps, ...rest } = props
  const { outlinedInput, ...restClasses } = classes

  if (variant === 'outlined') {
    rest.input = (
      <OutlinedInput
        {...inputProps}
        classes={{
          input: outlinedInput
        }}
      />
    )
  }

  return (
    <MUISelect
      classes={restClasses}
      {...inputProps}
      {...rest}
      MenuProps={{
        anchorOrigin: {
          vertical: 'bottom',
          horizontal: 'left'
        },
        transformOrigin: {
          vertical: 'top',
          horizontal: 'left'
        },
        getContentAnchorEl: undefined // needed to restore default behaviour
      }}
    >
      {renderOptions(props)}
    </MUISelect>
  )
}

Select.propTypes = {
  native: PropTypes.bool,
  onChange: PropTypes.func,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string,
      value: PropTypes.string.isRequired,
      text: PropTypes.oneOfType([PropTypes.node, PropTypes.string]).isRequired
    })
  ).isRequired
}

Select.defaultProps = {
  native: false,
  onChange: () => {}
}

export default withStyles(styles)(Select)
