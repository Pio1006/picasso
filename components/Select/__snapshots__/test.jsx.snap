// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dropdown select 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root MuiInput-underline"
  >
    <div
      class="MuiSelect-root"
    >
      <div
        aria-haspopup="true"
        aria-pressed="false"
        class="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        role="button"
        tabindex="0"
      >
        text1
      </div>
      <input
        type="hidden"
        value="val1"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-86 MuiSelect-icon-55"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`renders native select 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiInput-root MuiInput-underline"
  >
    <div
      class="MuiSelect-root"
    >
      <select
        class="MuiSelect-select MuiInputBase-input MuiInput-input"
      >
        <option
          value="val1"
        >
          text1
        </option>
        <option
          value="val2"
        >
          text2
        </option>
        <option
          value="val3"
        >
          text3
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-39 MuiSelect-icon-8"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
